pittgoogle/bigquery.py

```diff
@@ -1,692 +1,162 @@
-    in_cone = in_cone[0]
-
-    return in_cone
+    _table: Optional[bigquery.Table] = field(default=None, init=False)
+
+    @classmethod
+    def from_cloud(
+        cls,
+        name: str,
+        *,
+        dataset: Optional[str] = None,
+        survey: Optional[str] = None,
+        testid: Optional[str] = None,
+    ):
+        """Create a `Table` with a `client` using implicit credentials (no explicit `auth`).
+
+        The `projectid` will be retrieved from the `client`.
+
+        Parameters
+        ----------
+        name : `str`
+            Name of the table.
+        dataset : `str`, optional
+            Name of the dataset containing the table. Either this or a `survey` is required. If a
+            `testid` is provided, it will be appended to this name following the Pitt-Google naming syntax.
+        survey : `str`, optional
+            Name of the survey. This will be used as the name of the dataset if the `dataset` kwarg
+            is not provided. This kwarg is provided for convenience in cases where the Pitt-Google
+            naming syntax is used to name resources.
+        testid : `str`, optional
+            Pipeline identifier. If this is not `None`, `False`, or `"False"` it will be appended to
+            the dataset name. This is used in cases where the Pitt-Google naming syntax is used to name
+            resources. This allows pipeline modules to find the correct resources without interfering
+            with other pipelines that may have deployed resources with the same base names
+            (e.g., for development and testing purposes).
+        """
+        if dataset is None:
+            # [TODO] update the elasticc broker to name the dataset using the survey name only
+            dataset = survey
+        # if testid is not False, "False", or None, append it to the dataset
+        if testid and testid != "False":
+            dataset = f"{dataset}_{testid}"
+        client = bigquery.Client()
+        table = cls(name, dataset=dataset, projectid=client.project, client=client)
+        # make the get request now to create a connection to the table
+        _ = table.table
+        return table
+
+    @property
+    def auth(self) -> Auth:
+        """Credentials for the Google Cloud project that owns this topic.
+
+        This will be created from environment variables if `self._auth` is None.
+        """
+        if self._auth is None:
+            self._auth = Auth()
+
+        if (self.projectid != self._auth.GOOGLE_CLOUD_PROJECT) and (self.projectid is not None):
+            LOGGER.warning(f"setting projectid to match auth: {self._auth.GOOGLE_CLOUD_PROJECT}")
+        self.projectid = self._auth.GOOGLE_CLOUD_PROJECT
+
+        return self._auth
+
+    @property
+    def id(self) -> str:
+        """Fully qualified table ID."""
+        # make sure we have a projectid. if it needs to be set, call auth
+        if self.projectid is None:
+            self.auth
```

Comment:
Codacy is complaining about line 136 with "Statement seems to have no effect". It does have an effect, as noted in line 134, but I don't know whether this is an okay workflow by general coding standards. If there's no strong argument against it then I'll leave it, but if there is I can try to figure out how to do this differently.

Reply:
If it is okay to leave it, I suppose I should still add some logic to catch an error when the auth cannot be constructed  and give the user a helpful explanation. I can imagine a scenario where they just need to know the `id` and they can supply the `projectid` but don't have credentials for an `Auth` handy.

Reply:
I agree with Codacy's complaint.  It's not obvious that accessing the property calls something.  If it was at least `self.auth()` that would make more sense.